src/useFormState.js

```diff
@@ -140,18 +145,31 @@ export default function useFormState(initialState, options) {
         if (isCheckbox || isRadio) {
           return toString(ownValue);
         }
+
+        if (isRaw) {
+          // Return `undefined` if we don't have the value.
+          return formState.current.values[name];
+        }
+
         return hasValueInState ? formState.current.values[name] : '';
       },
       onChange: callbacks.getOrSet(ON_BLUR_HANDLER + key, e => {
         setDirty(name, true);
-        let { value } = e.target;
+        let value = isRaw ? e : e.target.value;
```

Comment:
Alright, tell me what you think.. I'd like to think this through carefully...

What do you think about allowing users to use an `onChange` callback when they use `raw()`?

```jsx
<CustomDateInput
 {raw({
   name: 'date',
   onChange: value => value.timestamp,
 })}
/>
```

So instead of always assuming the value will the object we get from `onChange`, users can decide the actual value they want to store in the state from the object being passed to `onChange`.

This would look something like this ([I have done something similar in #55](https://github.com/wsmd/react-use-form-state/blob/160b016ee1a8c531f7677b2293d65957810e891e/src/useFormState.js#L179)):

```js
// perhaps we could change the default onChange from a no-op to an identify
let value = isRaw ? formOptions.onChange(e) : e.target.value;

if (node.process.env === 'development') {
  if (value === undefined) {
    console.warn('hey, you used onChange with raw() but didn't return a value!')
  }
}
```

This also allows us to limit accepting a return value from onChange to `raw` inputs only.

Thoughts?